rotkehlchen/globaldb/handler.py

```diff
@@ -1203,6 +1227,144 @@ def get_historical_price_data(source: HistoricalPriceOracle) -> List[Dict[str, A
+            # be notified.
+            query = cursor.execute('SELECT identifier from assets;')
+            user_ids = set(query.fetchall())
+            # Attach to the clean db packaged with rotki
+            cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
+            # Get built in identifiers
+            query = cursor.execute('SELECT identifier from clean_db.assets;')
+            shipped_ids = set(query.fetchall())
+            diff_ids = user_ids - shipped_ids
+            if len(diff_ids) != 0:
+                # If we have non common elements we have to see if these assets are used anywhere
+                # in the database tables
+                diffs = ', '.join([f'"{id[0]}"' for id in diff_ids])
+                for table, column in TABLES_WITH_ASSETS:
+                    diffs_query = f'SELECT COUNT(*) FROM {table} WHERE {column} IN ({diffs})'
+                    user_db_cursor.execute(diffs_query)
+                    amount = user_db_cursor.fetchone()
+                    if amount[0] != 0:
+                        cursor.execute(detach_database)
+                        log.error(
+                            f'Assets inserted by the user are used in the database '
+                            f'tables {diff_ids}',
+                        )
+                        return False, f'There are assets that can not be deleted: {str(diff_ids)}'
+            # Check that versions match
+            query = cursor.execute('SELECT value from clean_db.settings WHERE name=="version";')
+            version = query.fetchone()
+            if int(version[0]) != _get_setting_value(cursor, 'version', GLOBAL_DB_VERSION):
+                cursor.execute(detach_database)
+                msg = (
+                    'Failed to restore assets. Global database is not '
+                    'updated to the latest version'
+                )
+                return False, msg
+            # If versions match drop tables
+            cursor.execute('PRAGMA foreign_keys = OFF;')
+            cursor.execute('DELETE FROM assets;')
+            cursor.execute('DELETE FROM ethereum_tokens;')
+            cursor.execute('DELETE FROM underlying_tokens_list;')
+            cursor.execute('DELETE FROM common_asset_details;')
+            # Copy assets
+            cursor.execute('INSERT INTO assets SELECT * FROM clean_db.assets;')
+            cursor.execute('INSERT INTO ethereum_tokens SELECT * FROM clean_db.ethereum_tokens;')
+            cursor.execute('INSERT INTO underlying_tokens_list SELECT * FROM clean_db.underlying_tokens_list;')  # noqa: E501
+            cursor.execute('INSERT INTO common_asset_details SELECT * FROM clean_db.common_asset_details;')  # noqa: E501
+            cursor.execute('PRAGMA foreign_keys = ON;')
+
+            user_db_cursor.execute('PRAGMA foreign_keys = OFF;')
+            user_db_cursor.execute('DELETE FROM assets;')
+            user_db_cursor.execute('PRAGMA foreign_keys = ON;')
```

Comment:
This foreign keys on should go AFTER the last insertion. The last insertion I see is in line 1302, right?

Even though I don't think that line can hit, still we should be safe and cleanly show where foreign keys are disable and enabled and for what reason

Reply:
you still have not done this.